File: boj/problem10818/main.py
from typing import List


def solve(nums: List[int]):
    max_num = -1000000
    min_num = 1000000

    for num in nums:
        min_num = min(num, min_num)
        max_num = max(num, max_num)

    return " ".join([str(min_num), str(max_num)])


def main():
    N = int(input())
    nums = map(int, input().split())

    print(solve(nums))


if __name__ == "__main__":
    main()

File: boj/problem10818/Main.java
package problem10818;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

public class Main {
    public static String solve(int[] nums) {
        int minNum = Integer.MAX_VALUE;
        int maxNum = Integer.MIN_VALUE;

        for (int num : nums) {
            minNum = Math.min(num, minNum);
            maxNum = Math.max(num, maxNum);
        }

        String answer = minNum + " " + maxNum;

        return answer;
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int N = Integer.parseInt(st.nextToken());
        st = new StringTokenizer(br.readLine());

        int[] nums = new int[N];
        for (int i = 0; i < N; i++) {
            nums[i] = Integer.parseInt(st.nextToken());
        }

        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        bw.write(solve(nums));
        bw.flush();
    }
}
